ratenyu/static/javascript/script.js

```diff
@@ -27,25 +27,176 @@ function resultCheckBoxClicked() {
+            dataList.removeChild(currentChild);
+    }
+}
+
+// Replaces options in the given dataList with newOptions
+function replaceDataListOptions(dataList, newOptions) {
+    // Add new options if they are not present
+    let existingOptions = [].slice.call(dataList.options).map((option) => option.value);
+    newOptions.forEach((optionValue) => {
+        if (!existingOptions.includes(optionValue)){
+            let newOption = document.createElement('option');
+            newOption.value = optionValue;
+            dataList.appendChild(newOption);
+        }
+    });
+
+    // Remove old options
+    filterOptions(dataList, newOptions);
+}
+
+// Custom form validation
+function validateForm(coursesData) {
+    let courseTitle = document.forms['add_review_form']['add_review_course_title'].value;
+    let courseId = document.forms['add_review_form']['add_review_course_id'].value;
+    let professorName = document.forms['add_review_form']['add_review_professor_name'].value;
+    let matchingCourseId = coursesData.find((course) => {
+        return course['display_course_id'] === courseId;
+    });
+    let matchingCourseTitle = coursesData.find((course) => {
+        return course['course_title'] === courseTitle;
+    });
+
+    // Validate Course Name, CourseID, and Course/Professor combo
+    if (!matchingCourseTitle) {
+        addErrorMessage('add-review-status',
+            'No Course found with Course Title "' + courseTitle + '."',
+            'no-course-title-found-message');
+        return false;
+    }
+    if (!matchingCourseId) {
+        addErrorMessage('add-review-status',
+            'No Course found with Course ID "' + courseId + '."',
+            'no-course-id-found-message');
+        return false;
+    } else {
+        if (matchingCourseTitle['display_course_id'] !== matchingCourseId['display_course_id']) {
+            addErrorMessage('add-review-status',
+                'No match found for Course Title "' + courseTitle + '" and Course ID "' + courseId + '."',
+                'no-course-title-id-match-message');
+            return false;
+        }
+        let professorOptions = matchingCourseId['professors'].map(value => value['professor_name']);
+        if (!professorOptions.includes(professorName)) {
+            addErrorMessage('add-review-status',
+                'No match found for Course ID "' + courseId + '" and Professor "' + professorName + '."',
+                'no-course-professor-match-message');
+            return false;
+        }
+    }
+
+    // Validate Rating requirement
+    return validateRatingRequirement();
+}
+
+function validateRatingRequirement() {
+    let star = document.querySelector('.my-star.add-star.star-1');
+    if (!star.classList.contains('is-active')) {
+        addErrorMessage('add-review-status',
+                'Please select a Review Rating.',
+                'no-rating-entered-message');
+            return false;
+    }
+    hideReviewForm();
```

Comment:
You don;'t need this. 
When i am not writing a review and clicking on submit, the form hides instead of showing error.